### ISSUE TITLE
plot(history) produces no output but works from console in RStudio

### ISSUE THREAD

**Issue by eafpres** [rstudio/keras#389]
Here is my code:

`              model <- keras_model_sequential() %>%
                layer_dense(units = (ncol(train_data) - 1), 
                            activation = activations[1], 
                            input_shape = c(ncol(train_data) - 1)) %>%
                layer_dropout(rate = 
                                if(dropout_passes == 1) {
                                  dropouts[1] 
                                } else {
                                  dropouts[dropout_index, 1]}) %>%
                layer_dense(units = 
                              if (unit_passes == 1) {
                                units_used[1]
                                } else {
                                  units_used[unit_index, 1]}, 
                            activation = activations[2]) %>%
                layer_dropout(rate = 
                                if(dropout_passes == 1) {
                                  dropouts[2] 
                                } else {
                                  dropouts[dropout_index, 2]}) %>%
                layer_dense(units = 
                              if (unit_passes == 1) {
                                units_used[2]
                              } else {
                                units_used[unit_index, 2]}, 
                            activation = activations[3]) %>%
                layer_dropout(rate = 
                                if(dropout_passes == 1) {
                                  dropouts[3] 
                                } else {
                                  dropouts[dropout_index, 3]}) %>%
                layer_dense(units = 
                              if (unit_passes == 1) {
                                units_used[3]
                              } else {
                                units_used[unit_index, 3]},
                            activation = activations[4]) %>%
                layer_dense(units = 1, activation = "linear")
#    
              model %>% compile(
                optimizer = 
                  optimizer_rmsprop(lr = 
                                      learning_rates[learning_rate_index], 
                                    decay = decays[decay_index]),
                loss = "mean_squared_error",
                metrics = "mean_absolute_error"
              )
#  
              history <- model %>% fit(
                x_train,
                y_train,
                view_metrics = FALSE,
                epochs = epochs_used[epoch_index],
                batch_size = batch_sizes_used[batch_size_index],
                validation_data = list(x_val, y_val)
              )
#
              plot(history)
              pred_bookings <- predict(model, rbind(x_train, x_val, x_test))
`
The plot(history) produces no plot output, but a later plot() of other data works fine.  However, if I use plot(history) from the console, it generates the plots OK.  If I set view_metrics = TRUE, I get the expected plots in the Viewer.

All this code is inside a loop (hence the indices).

**Comment by jjallaire** [comment]
Since it’s a ggplot you need to explicitly call print on it if it’s inside
a loop

**Comment by eafpres** [comment]
That worked.  Sorry for being ignorant!  Thank you for your help, and as well for your always prompt attention to everyone here.
